spec/lib/appsignal/hooks/action_cable_spec.rb

```diff
@@ -0,0 +1,300 @@
+describe Appsignal::Hooks::ActionCableHook do
+  if DependencyHelper.action_cable_present?
+    context "with ActionCable" do
+      require "action_cable/engine"
+
+      describe ".dependencies_present?" do
+        subject { described_class.new.dependencies_present? }
+
+        it "returns true" do
+          is_expected.to be_truthy
+        end
+      end
+
+      describe ActionCable::Channel::Base do
+        let(:transaction) do
+          Appsignal::Transaction.new(
+            request_id,
+            Appsignal::Transaction::ACTION_CABLE,
+            ActionDispatch::Request.new(env)
+          )
+        end
+        let(:channel) do
+          Class.new(ActionCable::Channel::Base) do
+            def speak(_data)
+            end
+
+            def self.to_s
+              "MyChannel"
+            end
+          end
+        end
+        let(:log) { StringIO.new }
+        let(:server) do
+          ActionCable::Server::Base.new.tap do |s|
+            s.config.logger = Logger.new(log)
+          end
+        end
+        let(:connection) { ActionCable::Connection::Base.new(server, env) }
+        let(:identifier) { { :channel => "MyChannel" }.to_json }
+        let(:params) { {} }
+        let(:request_id) { SecureRandom.uuid }
+        let(:env) do
+          http_request_env_with_data("action_dispatch.request_id" => request_id, :params => params)
+        end
+        let(:instance) { channel.new(connection, identifier, params) }
+        subject { transaction.to_h }
+        before do
+          start_agent
+          expect(Appsignal.active?).to be_truthy
+          transaction
+
+          expect(Appsignal::Transaction).to receive(:create)
+            .with(request_id, Appsignal::Transaction::ACTION_CABLE, kind_of(ActionDispatch::Request))
+            .and_return(transaction)
+          allow(Appsignal::Transaction).to receive(:current).and_return(transaction)
+          expect(transaction.ext).to receive(:complete) # and do nothing
+
+          # TODO: Nicer way to stub this without a websocket?
```

Comment:
Ah yes, that TODO. I wanted to fix that before submitting it, but looks like it slipped through. I'll have a looksy